### PR TITLE
do_Post bug fix and python3 related

### PR DESCRIPTION
**Pull request by mliang-mitre**
1. add requirements.txt for python venv
2. fix a bug in do_Post where a return statement is not indented correctly
3. numpy.int -> int to work with newer version of numpy
4. other changes to fix "empty response" issue when running fake-sas

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+attrs==23.2.0
+cffi==1.16.0
```
Comment (FW-ccaparanga) [inline requirements.txt:2]: What version of Python was used for generating this? Looks like cffi 1.16.0 dropped support for python 3.7, which is the stated version in the readme

Comment (mliang-mitre) [reply requirements.txt:2]: Python 3.11
Should I update readme to 3.11? Feels like that's the right thing to do, since 3.7 if end of life.

Comment (FW-ccaparanga) [reply requirements.txt:2]: Yes, please update the readme. I'll test this branch with python 3.11 to verify

Comment (mliang-mitre) [reply requirements.txt:2]: I've updated readme. Thank you for reviewing and testing.

**Review by FW-ccaparanga — APPROVED** [review]
I did some light validation to test out conda and venv + requirements.txt. Looks good.

**Comment by mliang-mitre** [comment]
Thank you @FW-ccaparanga for reviewing & testing.

**Pull request merged**